src/KafkaNetClient/ProtocolGateway.cs

```diff
@@ -37,8 +37,10 @@ public ProtocolGateway(KafkaOptions kafkaOptions)
         /// <exception cref="ResponseTimeoutException">Thrown if there request times out</exception>
         /// <exception cref="SocketException">Thrown in case of network error contacting broker (after retries)</exception>
         /// <exception cref="KafkaApplicationException">Thrown in case of an unexpected error in the request</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown in case of Topic is not valid</exception>
```

Comment:
it shouldn't be argument out of range exception, replace with a better suited one.
change comment to "Thrown in case the topic name is invalid"